static/js/timerender.js

```diff
@@ -311,12 +311,37 @@ export function get_full_datetime(time) {
     // The sign of getTimezoneOffset() is reversed wrt
     // the conventional meaning of UTC+n / UTC-n
     const tz_offset = -time.getTimezoneOffset() / 60;
+
+    // Since we'll be showing timezone in time string,
+    // (UTC+0) isn't much helpful while we show UTC.
+    const tz_offset_string =
+        tz_offset !== 0 ? " (UTC" + (tz_offset < 0 ? "-" : "+") + tz_offset + ")" : "";
+
+    const date_string_options = {weekday: "long", month: "long", day: "numeric"};
+    const time_string_options = {timeStyle: "long"};
+
+    if (page_params.twenty_four_hour_time) {
+        time_string_options.hourCycle = "h24";
+    }
```

Comment:
Fixes #18960

Reply:
Does this actually fix #18960?  I'd expect that issue to be a flatpickr configuration thing.

Reply:
oops, sorry. I miss read that. since this wasn't considering 24 hour time setting I probably assumed it's for this with a quick glance.